php/functionsIDD.php

```diff
@@ -189,6 +189,107 @@ function insertPerson($cnxn, $firstName, $lastName, $email, $phone) {
+                           VALUES ('$volunteerID', '$oneWeek', '$weekend')";
+
+    return mysqli_query($cnxn, $availabilityInsert);
+}
+
+function insertReference($cnxn, $personID, $volunteerID, $relationship){
+    $referenceInsert = "INSERT INTO reference (personId, volunteerId, relationship)
+                        VALUES ('$personID', '$volunteerID', '$relationship')";
+
+    return mysqli_query($cnxn, $referenceInsert);
+}
+
+function insertVolunteer($cnxn, $personID, $onMailList, $tShirtSize, $street, $city, $state, $zip, $heardAbout, $motivation, $experience, $skills, $joinDate){
+    $volunteerInsert = "INSERT INTO volunteer (personId, onMailList, tShirtSize, street, city, state, zip, heardAbout, motivation, experience, skills, joinDate)                                  
+                        VALUES ('$personID', '$onMailList', '$tShirtSize', '$street', '$city', '$state', '$zip', '$heardAbout', '$motivation', '$experience', '$skills', '$joinDate')";
+
+    return mysqli_query($cnxn, $volunteerInsert);
+}
+
+function saveVolunteer($cnxn, $firstName, $lastName, $email, $phone
+                            , $onMailList, $tShirtSize, $street, $city, $state, $zip, $heardAbout, $motivation, $experience, $skills,
+                            $refFirstArray, $refLastArray, $refPhoneArray, $refMailArray, $refRelationArray,
+                            $oneWeek, $weekends,
+                            $interestArray){
+
+
+    $personResult = insertPerson($cnxn, $firstName, $lastName, $email, $phone);
+
+    //checks if the insert into person was successful
+    if($personResult){
+        $personVolID = $cnxn->insert_id;
+        $currDate = date('Y-m-d');
+
+        $volunteerResult = insertVolunteer($cnxn, $personVolID, $onMailList, $tShirtSize, $street, $city, $state, $zip, $heardAbout, $motivation, $experience, $skills, $currDate);
+
+        //checks if the insert into volunteers was successful
+        if($volunteerResult){
+            $volunteerID = $cnxn->insert_id;
+
+            //inserts all three references. If any fail stops and sends false
+            for($i = 1; $i < 4; $i++){
+                $personResult = insertPerson($cnxn, $refFirstArray[$i], $refLastArray[$i], $refMailArray[$i], $refPhoneArray[$i]);
+                if($personResult === false){
+                    return false;
+                }
+
+                $personRefID = $cnxn->insert_id;
+
+                $referenceResult = insertReference($cnxn, $personRefID, $volunteerID, $refRelationArray[$i]);
+                if($referenceResult === false){
+                    return false;
+                }
+            }
+
+            $availableResult = insertAvailability($cnxn, $volunteerID, $oneWeek, $weekends);
+
+            if($availableResult){
+                for($i = 0; $i < count($interestArray); $i++){
+                    $interestResult = insertInterests($cnxn, $interestArray[$i]);
```

Comment:
The interest values except for what is typed into the "Other" text box is going to be the same every time. Inserting them every time is redundant data and defeats the purpose of having a joining table. We can manually insert the rows we want to set it up, and then when the user submits the form we just link the interests they chose in the volunteerInterest table. I think we have to do select queries to find out ids of Interest options. We only want to insert an interestOption in the interest table when it is a unique value (what the user types into the "Other" text input)